frontend/plugins/sales_ui/src/modules/deals/boards/components/common/GenericBoard.tsx

```
@@ -194,68 +296,70 @@ function GenericBoardInner<
+
+        if (finalState !== currentState) {
+          stateRef.current = finalState;
+          setState(finalState);
         }
-      }
 
-      onStateChange?.(state, initialState, draggedId);
+        onStateChange?.(finalState, initialState, draggedId);
+      }
     },
     // eslint-disable-next-line react-hooks/exhaustive-deps
```

ESLint

[error] Definition for rule 'react-hooks/exhaustive-deps' was not found.

(react-hooks/exhaustive-deps)